src/CodeStyle/CSharp/CodeFixes/CSharpFormattingAnalyzer.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpFormattingAnalyzer : AbstractFormattingAnalyzer
+    {
+        protected override Type GetAnalyzerImplType()
+        {
+            return typeof(CSharpFormattingAnalyzerImpl);
```

Comment:
could you explain what the design pattern is here?  What is the need for this sort of indirection?

Reply:
I need to ensure the assembly load handler is registered before the JIT sees anything that might need the workspaces layer.

Reply:
Comment added in a4a1961828c180ef059f79d9b2d610720b8fb6bb